internal/debug/debug.go

```diff
@@ -0,0 +1,32 @@
+package debug
+
+import (
+	"io"
+	"log"
+	"os"
+)
+
+var enabled = false
+var logger = log.New(os.Stderr, "DEBUG: ", log.LstdFlags|log.Lmsgprefix)
+
+// Printf works like log.Printf, but only when debugging is enabled.
+func Printf(format string, v ...interface{}) {
+	if enabled {
+		logger.Printf(format, v...)
+	}
+}
+
+// Println works like log.Println, but only when debugging is enabled.
+func Println(v ...interface{}) {
+	if enabled {
+		logger.Println(v...)
+	}
+}
+
+func SetDebug(b bool) {
+	enabled = b
+}
+
+func SetOutput(w io.Writer) {
```

Comment:
This is so that we can write log to files in the future. At the moment, it will output to stderr by default.

Reply:
I think we should codify that outputting to stdout by default instead of stderr is an explicit decision to conform to 12-factor principles.